src/bicycle/arend.cc

```diff
@@ -0,0 +1,154 @@
+    static constexpr auto steer_rate_index = index(full_state_index_t::steer_rate);
+
+    static constexpr auto yaw_index_A = index(state_index_t::yaw_angle);
+    static constexpr auto steer_index_A = index(state_index_t::steer_angle);
+    static constexpr auto steer_rate_index_A = index(state_index_t::steer_rate);
+
+    const state_matrix_t&A = m_A;
+    const real_t M_00 = m_M(0, 0);
+    const real_t C_01 = m_v*m_C1(0, 1);
+    const real_t K_01 = m_K(0, 1);
+    const real_t C_00 = m_v*m_C1(0, 0);
+    const real_t K_00 = m_K(0, 0);
+    const real_t v = m_v;
+    const real_t rr = m_rr;
+
+    const real_t steer_angle_measurement = z[index(output_index_t::steer_angle)];
+    const real_t steer_rate_measurement = z[index(output_index_t::steer_rate)];
+
+    full_state_t xout = xf;
+
+    m_stepper.do_step([&A, M_00, C_01, K_01, C_00, K_00, v, rr,
+                       steer_angle_measurement, steer_rate_measurement](
+                const full_state_t& x, full_state_t& dxdt, const real_t t) -> void {
+            (void)t; // system is time-independent;
+
+            // auxiliary state fields first
+            dxdt[x_index] = v*std::cos(x[yaw_index]);
+            dxdt[y_index] = v*std::sin(x[yaw_index]);
+            dxdt[rear_wheel_index] = -v/rr;
+            dxdt[pitch_index] = 0; // pitch angle is not integrated and must be obtained using solve_pitch_constraint()
+
+            // state fields
+            dxdt[yaw_index] = A(yaw_index_A, steer_index_A)*steer_angle_measurement +
+                              A(yaw_index_A, steer_rate_index_A)*steer_rate_measurement;
+            dxdt[roll_index] = x[roll_rate_index];
+            dxdt[steer_index] = 0; // disable steer angle integration
+            dxdt[roll_rate_index] = -(C_01*steer_rate_measurement +
+                                      K_01*steer_angle_measurement +
+                                      C_00*x[roll_rate_index] +
+                                      K_00*x[roll_index])/M_00;
+            dxdt[steer_rate_index] = 0; // disable steer rate integration
+            }, xout, static_cast<real_t>(0), t); // newly obtained state written in place
+    return xout;
+}
+
+void BicycleArend::set_state_space() {
+    Bicycle::set_state_space();
+    set_K();
+}
+
+void BicycleArend::set_output_element(output_t& x, output_index_t field, real_t value) {
+    x[index(field)] = value;
+}
+
+real_t BicycleArend::get_output_element(const output_t& x, output_index_t field) {
+    return x[index(field)];
+}
+
+BicycleArend::output_t BicycleArend::normalize_output(const output_t& y) const {
+    static_assert(index(output_index_t::steer_angle) == 0,
+        "Invalid underlying value for output index element");
+    static_assert(index(output_index_t::steer_rate) == 1,
+        "Invalid underlying value for output index element");
+    static_assert(index(output_index_t::number_of_types) == 2,
+        "Invalid underlying value for output index element");
```

Comment:
Where would you put them? I chose that location so I would remember to update that function if the enum class is changed.